nixos/modules/installer/tools/nixos-install.sh

```diff
@@ -131,24 +134,29 @@ touch "$mountPoint/etc/NIXOS"
 if [[ -z $noBootLoader ]]; then
     echo "installing the boot loader..."
     # Grub needs an mtab.
-    ln -sfn /proc/mounts $mountPoint/etc/mtab
+    ln -sfn /proc/mounts "$mountPoint/etc/mtab"
     NIXOS_INSTALL_BOOTLOADER=1 nixos-enter --root "$mountPoint" -- /run/current-system/bin/switch-to-configuration boot
 fi
 
 # Ask the user to set a root password, but only if the passwd command
 # exists (i.e. when mutable user accounts are enabled).
 if [[ -z $noRootPasswd ]] && [ -t 0 ]; then
     if nixos-enter --root "$mountPoint" -c 'test -e /nix/var/nix/profiles/system/sw/bin/passwd'; then
-        set +e
-        nixos-enter --root "$mountPoint" -c 'echo "setting root password..." && /nix/var/nix/profiles/system/sw/bin/passwd'
-        exit_code=$?
-        set -e
-
-        if [[ $exit_code != 0 ]]; then
-            echo "Setting a root password failed with the above printed error."
-            echo "You can set the root password manually by executing \`nixos-enter --root ${mountPoint@Q}\` and then running \`passwd\` in the shell of the new system."
-            exit $exit_code
-        fi
+        echo 'setting root password...'
+
+        # Ask the password until it succeeds or MAX_PASSWD_ATTEMPTS is reached
+        for ATTEMPT in $(seq 1 $MAX_PASSWD_ATTEMPTS); do
+          set +e
+          nixos-enter --root "$mountPoint" -c '/nix/var/nix/profiles/system/sw/bin/passwd' && break
```

Comment:
This always prints `setting up /etc...` from the activation script. I could imagine a `--silent` switch to `nixos-enter`, suppressing all activation script output, as this might be confusing if printed before a second `passwd` attempt. Don't know whether that's a good idea though... :neutral_face:

Reply:
That is probably coming from https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/system/etc/etc.nix#L156 not nixos-enter. I wonder why we even echo that.

Reply:
> That is probably coming from [...] not nixos-enter

Yes, hence the the proposal to silence all activation script output when running `nixos-enter`, which executes the activation script.

It is questionable whether `echo`ing that makes sense, but the activation script may print other stuff as well, so I still think a `--silent`-option to `nixos-enter` _could_ make sense anyways.

Reply:
It sure would not hurt, do you mind opening a PR for that?

Reply:
Can do. :+1: